### PR TITLE
[DOCS] Adds example of semantic search with ELSER

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds an example of how to use ELSER to perform semantic search.
The example uses a downloadable data set (the same that is used in the text embedding example) instead of bulk ingesting some documents to an index. The letter is easier, the former has more demonstrative power and provides more possibilities for the users to play around with the model and the functionality.

### Preview

[Semantic search with ELSER](https://elasticsearch_95992.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-elser.html)

### CODE REVIEW

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 1]
```diff
@@ -5,4 +5,221 @@
...
+
+[discrete]
+[[load-data]]
+=== Load data
+
+In this step, you load the data that you later use in the {infer} ingest 
+pipeline to extract tokens from it.
+
+NOTE: Only the first 512 extracted tokens of the ingested docs are considered 
+during semantic search.
```
Comment (droberts195) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:103]:
```suggestion
during semantic search with ELSER v1.
```

Comment (droberts195) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:103]:
```suggestion
during semantic search with ELSER v1.
```

**Review by droberts195 — APPROVED** [review]
LGTM

I think it's worth mentioning that the 512 token limit is for ELSER v1. Hopefully we'll remove that for the next version.

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 2]
```diff
@@ -5,4 +5,221 @@
...
+You can also open the Trained Models UI, select the Pipelines tab under ELSER to 
+follow the progress. It may take a couple of minutes to complete the process.
+
+
+[discrete]
+[[text-expansion-query]]
+=== Semantic search by using the `text_expansion` query
+
+To perform semantic search, pass a `text_expansion` object to the search API 
+call, and provide the query text and the ELSER model ID. The example below uses
```
Comment (abdonpijpelink) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:157]:
```suggestion
To perform semantic search, use the `text_expansion` query, 
and provide the query text and the ELSER model ID. The example below uses 
```

**Review by abdonpijpelink — APPROVED** [review]
LGTM. I left a minor suggestion.

General question: it wasn't clear to me initially that this was a tutorial. Maybe we should change the page title into "Tutorial: semantic search with ELSER"?

File: docs/reference/search/search-your-data/semantic-search-elser.asciidoc [hunk 3]
```diff
@@ -1,8 +1,225 @@
...
+    {
+      "inference": {
+        "model_id": ".elser_model_1",
+        "target_field": "ml",
+        "field_map": {
+          "text": "text_field"
+        },
+        "inference_config": {
+          "text_expansion": { <1>
+            "results_field": "ml.tokens"
```
Comment (szabosteve) [inline docs/reference/search/search-your-data/semantic-search-elser.asciidoc:82]:
```suggestion
            "results_field": "tokens"
```

**Comment by szabosteve** [comment]
@elasticsearchmachine test this please

**Review by davidkyle — APPROVED** [review]
Love the example query.

It would be good to add examples of using text_expansion in a compound (bool) query combining the query with a terms filter for example.

**Review by davidkyle — APPROVED** [review]
Love the example query.

It would be good to add examples of using text_expansion in a compound (bool) query combining the query with a terms filter for example.

**Comment by szabosteve** [comment]
@davidkyle I will add the combined example in another PR.

**Pull request merged**